src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -111,6 +153,57 @@ export class FileDataSource implements IDataSource {
 		}
 	}
 
+	private resolveVirtualDirectories(files: FileStat[]) {
+		files.forEach(x => x.isVirtualDirectory = x.isVirtualDirectoryMember = false);
+
+		if (!this.enableVirtualDirectories) {
+			return;
+		}
+
+		files = files.filter(x => !x.isDirectory);
+
+		let nesting = new Map<FileStat, FileStat[]>();
+		files.forEach(x => nesting.set(x, []));
+
+		files.filter(x => !x.isDirectory)
+			.map(x => ({
+				child: x,
+				parent: files.filter(y => x !== y && this.isVirtualDirectoryOf(y.name, x.name))[0]
+			}))
+			.filter(x => x.parent)
+			.forEach(({ parent, child }) => {
+				child.isVirtualDirectoryMember = true;
+				nesting.get(parent).push(child);
+			});
+
+		nesting.forEach((children, parent) => {
+			parent.children = children;
+			parent.isVirtualDirectory = !!children.length;
+		});
```

Comment:
So you execute it on `stat.parent.children` for virtualDirectory. It means that it looks only for files within same directory and nests them under. I cannot see how it nests files generated in different directory mentioned in [this comment](https://github.com/Microsoft/vscode/issues/6328#issuecomment-317594150). Am I missing here?